consensus/babe/babe.go

```diff
@@ -46,6 +47,7 @@ type Session struct {
 	txQueue        *tx.PriorityQueue
 	slotToProof    map[uint64]*VrfOutputAndProof // for slots where we are a producer, store the vrf output (bytes 0-32) + proof (bytes 32-96)
 	newBlocks      chan<- types.Block            // send blocks to core service
+	done           chan<- struct{}               // lets core know when the epoch is done
```

Comment:
Should this be it's own type? Why an empty struct?

Reply:
we don't need to send anything over the channel, just signal. empty struct is generally used in Go for this purpose since it's the smallest type we can have: sizeof(struct{}) == 0 since nothing is actually allocated)

Reply:
sweet, thanks for explaining